usaspending_api/common/query_with_filters.py

```diff
@@ -942,27 +983,27 @@ def _generate_elasticsearch_query(cls, filters: dict, query_type: _QueryType, **
         filters_copy = copy.deepcopy(filters)
 
         # tas_codes are unique in that the same query is spread across two keys
-        must_queries = cls._handle_tas_query(must_queries, filters_copy, query_type)
+        must_queries = self._handle_tas_query(must_queries, filters_copy, self.query_type)
```

Comment:
You should be able to update the `_handle_tas_query` method to be an instance method instead of a class method. And then that would also allow you to remove the `self.query_type` from the parameters as you could access it in the instance method.